docs/source/developer_guides/method_comparison/lora.md

```diff
@@ -0,0 +1,196 @@
+# LoRA (Low-Rank Adaptation)
+
+## Overview
+LoRA is a parameter-efficient fine-tuning method that introduces trainable low-rank matrices into transformer layers. It's particularly effective for large language models and offers a good balance between performance and resource efficiency.
+
+## Key Features
+- Memory efficient (~0.1% of base model parameters)
+- Minimal impact on inference speed
+- Easy to implement and use
+- Compatible with most transformer architectures
+
+## Performance Characteristics
+
+### Memory Efficiency
+| Model Size | LoRA Parameters | Memory Usage |
+|------------|----------------|--------------|
+| 1B         | ~1M            | ~4MB         |
+| 7B         | ~7M            | ~28MB        |
+| 13B        | ~13M           | ~52MB        |
+| 70B        | ~70M           | ~280MB       |
+
+### Training Performance
+| Metric | Value |
+|--------|-------|
+| Training Speed | Fast (similar to full fine-tuning) |
+| Convergence | Quick (typically 1-2 epochs) |
+| Inference Overhead | < 5% |
+
+## Use Cases
+
+### Best For
+- General fine-tuning tasks
+- Large language models
+- Multi-task learning
+- Resource-constrained environments
+
+### Not Recommended For
+- Tasks requiring extensive model modifications
+- Very small models (< 100M parameters)
+- Real-time applications with strict latency requirements
+
+## Implementation
+
+### Basic Usage
+```python
+from peft import LoraConfig, get_peft_model
+
+# Define LoRA configuration
+config = LoraConfig(
+    r=8,  # rank
+    lora_alpha=32,
+    target_modules=["q_proj", "v_proj"],
+    lora_dropout=0.05,
+    bias="none",
+)
+
+# Create PEFT model
+model = get_peft_model(model, config)
+```
+
+### Advanced Configuration
+```python
+# Custom LoRA configuration for specific needs
+config = LoraConfig(
+    r=16,  # higher rank for better performance
+    lora_alpha=64,
+    target_modules=["q_proj", "k_proj", "v_proj", "o_proj"],
+    lora_dropout=0.1,
+    bias="lora_only",
+    modules_to_save=["classifier"],
+)
+```
+
+## Hyperparameter Tuning
+
+### Recommended Ranges
+| Parameter | Recommended Range | Impact |
+|-----------|------------------|--------|
+| rank (r) | 4-32 | Higher = better performance, more parameters |
+| alpha | 8-64 | Controls scaling of LoRA weights |
```

Comment:
Let's mention that `alpha==2*rank` is a good starting point, as well as `use_rslora=True`.